rfcs/text/refactoring-date-component.md

```diff
@@ -195,36 +190,39 @@ Whilst we will continue to support the same custom event object, we will remove
     const [value, setValue] = useState("28/07/1987");
     const [hiddenValue, setHiddenValue] = useState("1987-07-28");
 
-    const updateDateValues = (event) => {
+    const handleChange = (event) => {
       const { formattedValue, rawValue } = event.target.value;
-      setVisibleValue(formattedValue);
-      setHiddenValue(rawValue)
+      setValue(formattedValue);
+      setHiddenValue(formatToIso(rawValue));
     };
 
+    const handleBlur = (event) => { 
+      // validation can be run here
+     }
+
     return (
-      <I18nProvider>
-        <div>
-          <Date
-            value={value}
-            onChange={updateValues}
-            onBlur={updateValues}
-            name="date-input"
-          />
-          <input
-            type="hidden"
-            name="date-input"
-            data-element="hidden-input"
-            value={hiddenValue}
-          />
-        </div>
-      </I18nProvider>
+      <>
+        <Date
+          value={value}
+          onChange={handleChange}
+          onBlur={handleBlur}
+          name="date-input"
+          picker={(pickerProps) => <Picker {...pickerProps}/>}
+        />
+        <input
+          type="hidden"
+          name="date-input"
+          data-element="hidden-input"
+          value={hiddenValue}
+        />
+      </>
     );
   };
 ```
 
 ## DatePicker component
 
-We will continue to use the same sub-components such as `Navbar` and `Weekday`, although they should be moved to an `__internal__` directory as will the `DatePicker` component as they will never need to be directly imported by consuming projects. The `containerProps` prop provides an interface for implementing keyboard accessibility within the picker (`tabIndex`, `onKeyDown` and so on). The [locale interface](https://react-day-picker.js.org/docs/localization) surfaced by `react-day-picker` will be integrated with whatever implementation we decide best meets requirements to override with the correct translations strings. Below is an example of how translating weekdays can be achieved using the locales surfaced from `date-fns`.
+We will continue to use the same sub-components such as `Navbar` and `Weekday`, although they should be moved to an `__internal__` directory of the `Picker` component as they will never need to be directly imported by consuming projects. The `containerProps` prop provides an interface for implementing keyboard accessibility within the picker (`tabIndex`, `onKeyDown` and so on). The [locale interface](https://react-day-picker.js.org/docs/localization) surfaced by `react-day-picker` will be integrated with `date-fns/locale` to override the translations strings. Below is an example of how translating weekdays can be achieved using the locales surfaced from `date-fns/locale`. It is also possible to generate an array of `months` based on the locale string.
```

Comment:
It's not clear why we need a `Picker` component if we're using a render prop. Is this so we can provide our own components that `react-day-picker` then displays?

Reply:
Yeah my intention was to surface our `Picker` which will be composed of components that are then passed into the `react-day-picker`, it will also allow us to include the translations and locale implementation as part of this component rather than having the logic in the main `Date` component and passing it down as props to the `Picker`. Of course having it as a render prop will mean someone could pass their own picker (even using an alternative to `react-day-picker` if they want) in as long as it matches the interface the input expects, this will mean they can deliver any features we don't want to add or not to have a picker at all and just have the input. I still envision that most of the time implementation teams will be using a `Picker` that is similar to the one here so think it's right that we export one to avoid forcing teams to write the same code and install the dependencies required for it etc.